Problem_Set_4/ps4a.py

```diff
@@ -0,0 +1,64 @@
+# Problem Set 4A
+# Name: <your name here>
+# Collaborators:
+# Time Spent: x:xx
+
+def get_permutations(sequence):
+    '''
+    Enumerate all permutations of a given string
+
+    sequence (string): an arbitrary string to permute. Assume that it is a
+    non-empty string.  
+
+    You MUST use recursion for this part. Non-recursive solutions will not be
+    accepted.
+
+    Returns: a list of all permutations of sequence
+
+    Example:
+    >>> get_permutations('abc')
+    ['abc', 'acb', 'bac', 'bca', 'cab', 'cba']
+
+    Note: depending on your implementation, you may return the permutations in
+    a different order than what is listed here.
+    '''
+    permutations = []
+
+    def insert(sequence, letter):

```

Comment:
You're defining a function inside another function?

Reply:
yeah I saw them do it in one of the lectures. I had a hard time figuring out what to do here, thing started working and defining another function made everything look neater.

Reply:
I mean you can still just define the function outside of this. Your call either way I guess